cmd/openshift-install/create.go

```diff
@@ -262,8 +270,64 @@ func destroyBootstrap(ctx context.Context, directory string) (err error) {
 	return destroybootstrap.Destroy(rootOpts.dir)
 }
 
+// waitForconsole
+func waitForConsole(ctx context.Context, directory string) (string, error) {
+	url := ""
+	// Need to keep these updated if they change
+	consoleNamespace := "openshift-console"
+	consoleRouteName := "console"
+	config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(directory, "auth", "kubeconfig"))
+	if err != nil {
+		return "", errors.Wrap(err, "loading kubeconfig")
+	}
+
+	rc, err := routeclient.NewForConfig(config)
+	if err != nil {
+		return "", errors.Wrap(err, "creating a route client")
+	}
+
+	consoleRouteTimeout := 10 * time.Minute
+	logrus.Infof("Waiting %v for the openshift-console route to be created...", consoleRouteTimeout)
+	consoleRouteContext, cancel := context.WithTimeout(ctx, consoleRouteTimeout)
+	defer cancel()
+	// Poll quickly but only log when the response
+	// when we've seen 15 of the same errors or output of
+	// no route in a row (to show we're still alive).
+	logDownsample := 15
+	silenceRemaining := logDownsample
+	wait.Until(func() {
+		consoleRoutes, err := rc.RouteV1().Routes(consoleNamespace).List(metav1.ListOptions{})
+		if err == nil && len(consoleRoutes.Items) > 0 {
+			for _, route := range consoleRoutes.Items {
+				logrus.Debugf("Route found in openshift-console namespace: %s\n", route.Name)
+				if route.Name == consoleRouteName {
+					url = fmt.Sprintf("https://%s", route.Spec.Host)
+				}
+			}
+			logrus.Debug("OpenShift console route is created")
```

Comment:
We should probably error out if `uri == ""` still by this point.

Reply:
done